### PR TITLE
Fix infinite reconcile loop

### PR DESCRIPTION
**Pull request by leifmadsen**
Fix a potential infinite reconcile loop when deploying Grafana due to
updated hashing algorithm which changes the hash contents each time the
contents is hashed. Adds a fix which results in skipping the update of
the default-grafana-htpasswd Secret object to avoid the situation.

Also makes a small add to the when clause list for deploying
Elasticsearch so that only when it is enabled are the tasks included.

Closes: STF-1499
Signed-off-by: Leif Madsen <lmadsen@redhat.com>

### CODE REVIEW

**Comment by leifmadsen** [comment]
I've done several tests now and I consider this ready for review.

The primary issue here was that when enabling hybrid mode in order to enable grafana, I was running into an infinite reconcile loop in STO due to the data in the default-grafana-htpasswd continually updating.

The changes here adds the same password hash in annotation to the secret, the same as how we do that for the deployment of grafana in order to allow restarting the grafana pod when the password is changed.

In this setup, I add a when clause to check that the annotation and annotation.hash-of-creds[...] is added for existing deployments (STF 1.5.2 and earlier), along with doing a secret update when the password hash changes (due to change of adminPassword in ServiceTelemetry CR), which happens just prior to the update of the deployment manifest for grafana so that the password is re-read by the grafana pod restart.

I am now able to flip between observabilityStrategy use_redhat and use_hybrid to extend the OBO deployment, and allowing for grafana to be deployed.

There are also a couple other efficiency changes added so that we're not creating secrets for elasticsearch when that is not enabled.

File: roles/servicetelemetry/vars/dummy_user_certs.yml [hunk 1]
```diff
@@ -53,4 +53,4 @@ elastic_user_key_dummy: |
   njxhAQKBgQD0lOpKtL8qz9gmqtkhDRe+EPHSX8rfirqqRrPUiwK7kAJeW2vtU8aa
   hFT7lEDjb7ERyZfybIkTVVBipKx2yse9nE+1dPGIgZop3E1guDuF9aOAzIUd/+/s
   CI7s/lIBZsPD3PyxXXRtsvN7iUv5tLvNFhfomB7miTYHE+MC5QHJVQ==
-  -----END RSA PRIVATE KEY-----
\ No newline at end of file
+  -----END RSA PRIVATE KEY-----
```
Comment (leifmadsen) [inline roles/servicetelemetry/vars/dummy_user_certs.yml:56]: this just fixes a local ansible-lint complaint about missing newline

**Comment by leifmadsen** [comment]
test

**Review by csibbitt — CHANGES_REQUESTED** [review]
The code here looks okay, but I think it decreases security. Storing the password twice with two hashes increases the attack surface, and using a global salt for one of them somewhat weakens any hashes that share that global salt. I also think it could be simpler.

The root of the problem is that `password_hash("bcrypt")` uses a random salt (no second parameter), and the proposed solution is to store the creds a second time with a static salt by using `password_hash('sha256', (session_secret | b64encode)[:16] )` (note the second parameter). Then you use the now-static hash to detect changes. 

Simply using a static (not necesarrily global!) salt with bcrypt should solve the problem without extra gymnastics.

The best solution is to use a random salt when first generating the secret, but if the secret already exists, you reuse the existing salt (the 22 chars after the third $, see [3] Below) so that it's deterministic. That way we are still using a random, non-global salt, but each password's salt remains static through its lifetime, preventing the hash mutation and infinite loop problem.

[1] https://docs.ansible.com/ansible/latest/collections/ansible/builtin/password_lookup.html
[2] https://passlib.readthedocs.io/en/stable/lib/passlib.hash.bcrypt.html
[3] https://en.wikipedia.org/wiki/Bcrypt#Description

## Below
```
$ ansible localhost -m ansible.builtin.debug -a "msg=\"{{ 'test'| password_hash('bcrypt') }}\""
    "msg": "$2b$12$kFm89kOb8wJ.UzKHXk6PbOYNNA14Xwu2Jyd7RoT7rn4LN5eQq2Tam"
            \__/\/ \____________________/\_____________________________/
            Alg Cost      Salt                        Hash

# Random salt -> different string
$ ansible localhost -m ansible.builtin.debug -a "msg=\"{{ 'test'| password_hash('bcrypt') }}\""
    "msg": "$2b$12$OzXVBHsQlpeHX.dAPIeEm.xaIvVE9UjWvJI8RaItQfRI6vrj9hDbG"

# Original salt -> original string
$ ansible localhost -m ansible.builtin.debug -a "msg=\"{{ 'test'| password_hash('bcrypt', 'kFm89kOb8wJ.UzKHXk6PbO') }}\""
    "msg": "$2b$12$kFm89kOb8wJ.UzKHXk6PbOYNNA14Xwu2Jyd7RoT7rn4LN5eQq2Tam"
```

**Comment by leifmadsen** [comment]
Read comments from Chris and talked with him out-of-band to come up with a proof that I can use to simplify everything. I'll remove the annotation from the secret, and can migrate from using the old hashing method to using bcrypt-based hashing everywhere rather than currently used sha256 for the grafana deployment (used for re-triggering deployment on password change).

```yaml
---
- name: k8s test
  hosts: localhost

  tasks:
  - name: Check for existing grafana htpasswd secret
    no_log: true
    k8s_info:
      api_version: v1
      kind: Secret
      namespace: service-telemetry
      name: 'default-grafana-htpasswd'
    register: grafana_htpasswd_secret

  - debug:
      var: grafana_htpasswd_secret.resources[0].data.auth | b64decode

  - set_fact:
      grafana_htpasswd_salt: "{{ (grafana_htpasswd_secret.resources[0].data.auth | b64decode | split('$') | last)[0:22] }}"

  - debug:
      var: grafana_htpasswd_salt

  - set_fact:
      grafana_secret_contents: "{{ ('supersecret' | password_hash('bcrypt', grafana_htpasswd_salt, ident='2y')) }}"

  - debug:
      var: grafana_secret_contents[29:]
```

**Output**

```bash
$  ansible-playbook -i localhost test.yml
[WARNING]: Unable to parse /home/leif/src/github.com/infrawatch/service-telemetry-operator/localhost as an inventory source
[WARNING]: No inventory was parsed, only implicit localhost is available
[WARNING]: provided hosts list is empty, only localhost is available. Note that the implicit localhost does not match 'all'

PLAY [k8s test] ************************************************************************************************************************************************************

TASK [Gathering Facts] *****************************************************************************************************************************************************
ok: [localhost]

TASK [Check for existing grafana htpasswd secret] **************************************************************************************************************************
ok: [localhost]

TASK [debug] ***************************************************************************************************************************************************************
ok: [localhost] => {
    "grafana_htpasswd_secret.resources[0].data.auth | b64decode": "root:$2y$12$IijTu/iS92SgPqBDkgU0.OLrilpryAx4b9lDCc6DzeO0nRwDW09LC"
}

TASK [set_fact] ************************************************************************************************************************************************************
ok: [localhost]

TASK [debug] ***************************************************************************************************************************************************************
ok: [localhost] => {
    "grafana_htpasswd_salt": "IijTu/iS92SgPqBDkgU0.O"
}

TASK [set_fact] ************************************************************************************************************************************************************
ok: [localhost]

TASK [debug] ***************************************************************************************************************************************************************
ok: [localhost] => {
    "grafana_secret_contents[29:]": "LrilpryAx4b9lDCc6DzeO0nRwDW09LC"
}

PLAY RECAP *****************************************************************************************************************************************************************
localhost                  : ok=7    changed=0    unreachable=0    failed=0    skipped=0    rescued=0    ignored=0   
```

**Comment by leifmadsen** [comment]
Latest set of changes. I will remove the debug logs etc from the uploaded PR, but here is the output from logs.

First iteration with no Secret created. This is done with Grafana disabled to start, but secrets are made anyways.

```bash
--------------------------- Ansible Task StdOut -------------------------------

TASK [servicetelemetry : Check for existing grafana htpasswd secret] ***********
task path: /opt/ansible/roles/servicetelemetry/tasks/component_grafana.yml:10

-------------------------------------------------------------------------------
{"level":"info","ts":1692986575.8124166,"logger":"logging_event_handler","msg":"[playbook task start]","name":"default","namespace":"service-telemetry","gvk":"infra.watch/v1beta1, Kind=ServiceTelemetry","event_type":"playbook_on_task_start","job":"5755568030158739278","EventData.Name":"servicetelemetry : Check for existing grafana htpasswd secret"}
{"level":"info","ts":1692986576.657023,"logger":"proxy","msg":"Cache miss: /v1, Kind=Secret, service-telemetry/default-grafana-htpasswd"}

--------------------------- Ansible Task StdOut -------------------------------

 TASK [debug] ******************************** 
ok: [localhost] => {
    "grafana_htpasswd_secret": {
        "changed": false,
        "failed": false,
        "resources": []
    }
}

-------------------------------------------------------------------------------
{"level":"info","ts":1692986576.8599453,"logger":"logging_event_handler","msg":"[playbook debug]","name":"default","namespace":"service-telemetry","gvk":"infra.watch/v1beta1, Kind=ServiceTelemetry","event_type":"runner_on_ok","job":"5755568030158739278","EventData.TaskArgs":""}

--------------------------- Ansible Task StdOut -------------------------------

 TASK [debug] ******************************** 
ok: [localhost] => {
    "msg": "Enter the block"
}

-------------------------------------------------------------------------------
{"level":"info","ts":1692986576.8762813,"logger":"logging_event_handler","msg":"[playbook debug]","name":"default","namespace":"service-telemetry","gvk":"infra.watch/v1beta1, Kind=ServiceTelemetry","event_type":"runner_on_ok","job":"5755568030158739278","EventData.TaskArgs":""}

--------------------------- Ansible Task StdOut -------------------------------

 TASK [Set current Grafana htpasswd salt] ******************************** 
fatal: [localhost]: FAILED! => {"msg": "The task includes an option with an undefined variable. The error was: list object has no element 0\n\nThe error appears to be in '/opt/ansible/roles/servicetelemetry/tasks/component_grafana.yml': line 26, column 7, but may\nbe elsewhere in the file depending on the exact syntax problem.\n\nThe offending line appears to be:\n\n\n    - name: Set current Grafana htpasswd salt\n      ^ here\n"}

-------------------------------------------------------------------------------
{"level":"error","ts":1692986577.0040116,"logger":"logging_event_handler","msg":"","name":"default","namespace":"service-telemetry","gvk":"infra.watch/v1beta1, Kind=ServiceTelemetry","event_type":"runner_on_failed","job":"5755568030158739278","EventData.Task":"Set current Grafana htpasswd salt","EventData.TaskArgs":"","EventData.FailedTaskPath":"/opt/ansible/roles/servicetelemetry/tasks/component_grafana.yml:26","error":"[playbook task failed]"}

--------------------------- Ansible Task StdOut -------------------------------

 TASK [debug] ******************************** 
ok: [localhost] => {
    "msg": "Enter rescue"
}

-------------------------------------------------------------------------------
{"level":"info","ts":1692986577.019443,"logger":"logging_event_handler","msg":"[playbook debug]","name":"default","namespace":"service-telemetry","gvk":"infra.watch/v1beta1, Kind=ServiceTelemetry","event_type":"runner_on_ok","job":"5755568030158739278","EventData.TaskArgs":""}

--------------------------- Ansible Task StdOut -------------------------------

 TASK [debug] ******************************** 
ok: [localhost] => {
    "init_grafana_htpasswd_bcrypt_string": "$2y$12$28n2si2pRsfY1Dr0UQOLG.QXZIadRbI5IET.dgndJgKQxdCYTg55C"
}

-------------------------------------------------------------------------------
{"level":"info","ts":1692986577.6068509,"logger":"logging_event_handler","msg":"[playbook debug]","name":"default","namespace":"service-telemetry","gvk":"infra.watch/v1beta1, Kind=ServiceTelemetry","event_type":"runner_on_ok","job":"5755568030158739278","EventData.TaskArgs":""}

--------------------------- Ansible Task StdOut -------------------------------

 TASK [debug] ******************************** 
ok: [localhost] => {
    "grafana_htpasswd_salt": "28n2si2pRsfY1Dr0UQOLG."
}

-------------------------------------------------------------------------------
{"level":"info","ts":1692986577.8788757,"logger":"logging_event_handler","msg":"[playbook debug]","name":"default","namespace":"service-telemetry","gvk":"infra.watch/v1beta1, Kind=ServiceTelemetry","event_type":"runner_on_ok","job":"5755568030158739278","EventData.TaskArgs":""}

--------------------------- Ansible Task StdOut -------------------------------

 TASK [debug] ******************************** 
ok: [localhost] => {
    "msg": "Enter always"
}

-------------------------------------------------------------------------------
{"level":"info","ts":1692986577.8965406,"logger":"logging_event_handler","msg":"[playbook debug]","name":"default","namespace":"service-telemetry","gvk":"infra.watch/v1beta1, Kind=ServiceTelemetry","event_type":"runner_on_ok","job":"5755568030158739278","EventData.TaskArgs":""}

--------------------------- Ansible Task StdOut -------------------------------

 TASK [debug] ******************************** 
ok: [localhost] => {
    "grafana_htpasswd_bcrypt_string": "$2y$12$28n2si2pRsfY1Dr0UQOLG.QXZIadRbI5IET.dgndJgKQxdCYTg55C"
}

-------------------------------------------------------------------------------
{"level":"info","ts":1692986578.5206218,"logger":"logging_event_handler","msg":"[playbook debug]","name":"default","namespace":"service-telemetry","gvk":"infra.watch/v1beta1, Kind=ServiceTelemetry","event_type":"runner_on_ok","job":"5755568030158739278","EventData.TaskArgs":""}

--------------------------- Ansible Task StdOut -------------------------------

 TASK [debug] ******************************** 
ok: [localhost] => {
    "grafana_htpasswd_auth_string": "root:$2y$12$28n2si2pRsfY1Dr0UQOLG.QXZIadRbI5IET.dgndJgKQxdCYTg55C"
}

-------------------------------------------------------------------------------
{"level":"info","ts":1692986578.7997184,"logger":"logging_event_handler","msg":"[playbook debug]","name":"default","namespace":"service-telemetry","gvk":"infra.watch/v1beta1, Kind=ServiceTelemetry","event_type":"runner_on_ok","job":"5755568030158739278","EventData.TaskArgs":""}

--------------------------- Ansible Task StdOut -------------------------------

TASK [servicetelemetry : Create or patch htpasswd secret for grafana admin] ****
task path: /opt/ansible/roles/servicetelemetry/tasks/component_grafana.yml:72

-------------------------------------------------------------------------------
{"level":"info","ts":1692986578.8047097,"logger":"logging_event_handler","msg":"[playbook task start]","name":"default","namespace":"service-telemetry","gvk":"infra.watch/v1beta1, Kind=ServiceTelemetry","event_type":"playbook_on_task_start","job":"5755568030158739278","EventData.Name":"servicetelemetry : Create or patch htpasswd secret for grafana admin"}
{"level":"info","ts":1692986579.6266491,"logger":"proxy","msg":"Cache miss: /v1, Kind=Secret, service-telemetry/default-grafana-htpasswd"}
{"level":"info","ts":1692986579.6300638,"logger":"proxy","msg":"Injecting owner reference"}
{"level":"info","ts":1692986579.6302788,"logger":"proxy","msg":"Watching child resource","kind":"/v1, Kind=Secret","enqueue_kind":"infra.watch/v1beta1, Kind=ServiceTelemetry"}
{"level":"info","ts":1692986579.6303108,"logger":"controller.servicetelemetry-controller","msg":"Starting EventSource","source":"kind source: /v1, Kind=Secret"}

--------------------------- Ansible Task StdOut -------------------------------

 TASK [Lookup template] ******************************** 
ok: [localhost] => {
    "msg": {
        "apiVersion": "integreatly.org/v1alpha1",
        "kind": "Grafana",
        "metadata": {
            "name": "default",
            "namespace": "service-telemetry"
        },
        "spec": {
            "baseImage": "docker.io/grafana/grafana:8.1.2",
            "client": {
                "preferService": true
            },
            "config": {
                "auth": {
                    "disable_login_form": true,
                    "disable_signout_menu": false
                },
                "auth.anonymous": {
                    "enabled": true,
                    "org_role": "Admin"
                },
                "log": {
                    "level": "warn",
                    "mode": "console"
                }
            },
            "containers": [
                {
                    "args": [
                        "-provider=openshift",
                        "-pass-basic-auth=false",
                        "-https-address=:3002",
                        "-htpasswd-file=/etc/proxy/htpasswd/auth",
                        "-tls-cert=/etc/tls/private/tls.crt",
                        "-tls-key=/etc/tls/private/tls.key",
                        "-upstream=http://localhost:3000",
                        "-cookie-secret-file=/etc/proxy/secrets/session_secret",
                        "-openshift-service-account=grafana-serviceaccount",
                        "-openshift-sar={\"resource\": \"namespaces\", \"verb\": \"get\"}",
                        "-openshift-ca=/etc/pki/tls/cert.pem",
                        "-openshift-ca=/var/run/secrets/kubernetes.io/serviceaccount/ca.crt"
                    ],
                    "image": "quay.io/openshift/origin-oauth-proxy:latest",
                    "name": "oauth-proxy",
                    "ports": [
                        {
                            "containerPort": 3002,
                            "name": "https",
                            "protocol": "TCP"
                        }
                    ],
                    "volumeMounts": [
                        {
                            "mountPath": "/etc/tls/private",
                            "name": "secret-default-grafana-proxy-tls"
                        },
                        {
                            "mountPath": "/etc/proxy/secrets",
                            "name": "secret-default-session-secret"
                        },
                        {
                            "mountPath": "/etc/proxy/htpasswd",
                            "name": "secret-default-grafana-htpasswd"
                        }
                    ]
                }
            ],
            "dashboardLabelSelector": [
                {
                    "matchExpressions": [
                        {
                            "key": "app",
                            "operator": "In",
                            "values": [
                                "grafana"
                            ]
                        }
                    ]
                }
            ],
            "deployment": {
                "annotations": {
                    "hash-of-creds-to-force-restart-if-changed": "cm9vdDokMnkkMTIkMjhuMnNpMnBSc2ZZMURyMFVRT0xHLlFYWklhZFJiSTVJRVQuZGduZEpnS1F4ZENZVGc1NUM="
                }
            },
            "ingress": {
                "enabled": true,
                "targetPort": "web",
                "termination": "reencrypt"
            },
            "secrets": [
                "default-grafana-proxy-tls",
                "default-session-secret",
                "default-grafana-htpasswd"
            ],
            "service": {
                "annotations": {
                    "service.alpha.openshift.io/serving-cert-secret-name": "default-grafana-proxy-tls"
                },
                "ports": [
                    {
                        "name": "web",
                        "port": 3002,
                        "protocol": "TCP",
                        "targetPort": "https"
                    }
                ]
            },
            "serviceAccount": {
                "annotations": {
                    "serviceaccounts.openshift.io/oauth-redirectreference.primary": "{\"kind\": \"OAuthRedirectReference\", \"apiVersion\": \"v1\", \"reference\": {\"kind\": \"Route\", \"name\": \"grafana-route\"}}"
                }
            }
        }
    }
}

-------------------------------------------------------------------------------
```

Then the 2nd reconcile loop which stops.

```bash
--------------------------- Ansible Task StdOut -------------------------------

TASK [servicetelemetry : Check for existing grafana htpasswd secret] ***********
task path: /opt/ansible/roles/servicetelemetry/tasks/component_grafana.yml:10

-------------------------------------------------------------------------------
{"level":"info","ts":1692986588.5351532,"logger":"logging_event_handler","msg":"[playbook task start]","name":"default","namespace":"service-telemetry","gvk":"infra.watch/v1beta1, Kind=ServiceTelemetry","event_type":"playbook_on_task_start","job":"6320470104503553430","EventData.Name":"servicetelemetry : Check for existing grafana htpasswd secret"}
{"level":"info","ts":1692986589.27782,"logger":"proxy","msg":"Read object from cache","resource":{"IsResourceRequest":true,"Path":"/api/v1/namespaces/service-telemetry/secrets/default-grafana-htpasswd","Verb":"get","APIPrefix":"api","APIGroup":"","APIVersion":"v1","Namespace":"service-telemetry","Resource":"secrets","Subresource":"","Name":"default-grafana-htpasswd","Parts":["secrets","default-grafana-htpasswd"]}}

--------------------------- Ansible Task StdOut -------------------------------

 TASK [debug] ******************************** 
ok: [localhost] => {
    "grafana_htpasswd_secret": {
        "changed": false,
        "failed": false,
        "resources": [
            {
                "apiVersion": "v1",
                "data": {
                    "auth": "cm9vdDokMnkkMTIkMjhuMnNpMnBSc2ZZMURyMFVRT0xHLlFYWklhZFJiSTVJRVQuZGduZEpnS1F4ZENZVGc1NUM="
                },
                "kind": "Secret",
                "metadata": {
                    "creationTimestamp": "2023-08-25T18:02:59Z",
                    "managedFields": [
                        {
                            "apiVersion": "v1",
                            "fieldsType": "FieldsV1",
                            "fieldsV1": {
                                "f:data": {
                                    ".": {},
                                    "f:auth": {}
                                },
                                "f:metadata": {
                                    "f:ownerReferences": {
                                        ".": {},
                                        "k:{\"uid\":\"275e4a41-efca-47c1-ac4d-ece905eac27b\"}": {}
                                    }
                                },
                                "f:type": {}
                            },
                            "manager": "OpenAPI-Generator",
                            "operation": "Update",
                            "time": "2023-08-25T18:02:59Z"
                        }
                    ],
                    "name": "default-grafana-htpasswd",
                    "namespace": "service-telemetry",
                    "ownerReferences": [
                        {
                            "apiVersion": "infra.watch/v1beta1",
                            "kind": "ServiceTelemetry",
                            "name": "default",
                            "uid": "275e4a41-efca-47c1-ac4d-ece905eac27b"
                        }
                    ],
                    "resourceVersion": "44279327",
                    "uid": "66167927-74f8-4bbe-bd4d-6d116547f8a1"
                },
                "type": "Opaque"
            }
        ]
    }
}

-------------------------------------------------------------------------------
{"level":"info","ts":1692986589.5137405,"logger":"logging_event_handler","msg":"[playbook debug]","name":"default","namespace":"service-telemetry","gvk":"infra.watch/v1beta1, Kind=ServiceTelemetry","event_type":"runner_on_ok","job":"6320470104503553430","EventData.TaskArgs":""}

--------------------------- Ansible Task StdOut -------------------------------

 TASK [debug] ******************************** 
ok: [localhost] => {
    "msg": "Enter the block"
}

-------------------------------------------------------------------------------
{"level":"info","ts":1692986589.5341961,"logger":"logging_event_handler","msg":"[playbook debug]","name":"default","namespace":"service-telemetry","gvk":"infra.watch/v1beta1, Kind=ServiceTelemetry","event_type":"runner_on_ok","job":"6320470104503553430","EventData.TaskArgs":""}

--------------------------- Ansible Task StdOut -------------------------------

 TASK [debug] ******************************** 
ok: [localhost] => {
    "grafana_htpasswd_salt": "28n2si2pRsfY1Dr0UQOLG."
}

-------------------------------------------------------------------------------
{"level":"info","ts":1692986589.7715871,"logger":"logging_event_handler","msg":"[playbook debug]","name":"default","namespace":"service-telemetry","gvk":"infra.watch/v1beta1, Kind=ServiceTelemetry","event_type":"runner_on_ok","job":"6320470104503553430","EventData.TaskArgs":""}

--------------------------- Ansible Task StdOut -------------------------------

 TASK [debug] ******************************** 
ok: [localhost] => {
    "msg": "Enter always"
}

-------------------------------------------------------------------------------
{"level":"info","ts":1692986589.78919,"logger":"logging_event_handler","msg":"[playbook debug]","name":"default","namespace":"service-telemetry","gvk":"infra.watch/v1beta1, Kind=ServiceTelemetry","event_type":"runner_on_ok","job":"6320470104503553430","EventData.TaskArgs":""}

--------------------------- Ansible Task StdOut -------------------------------

 TASK [debug] ******************************** 
ok: [localhost] => {
    "grafana_htpasswd_bcrypt_string": "$2y$12$28n2si2pRsfY1Dr0UQOLG.QXZIadRbI5IET.dgndJgKQxdCYTg55C"
}

-------------------------------------------------------------------------------
{"level":"info","ts":1692986590.4312983,"logger":"logging_event_handler","msg":"[playbook debug]","name":"default","namespace":"service-telemetry","gvk":"infra.watch/v1beta1, Kind=ServiceTelemetry","event_type":"runner_on_ok","job":"6320470104503553430","EventData.TaskArgs":""}

--------------------------- Ansible Task StdOut -------------------------------

 TASK [debug] ******************************** 
ok: [localhost] => {
    "grafana_htpasswd_auth_string": "root:$2y$12$28n2si2pRsfY1Dr0UQOLG.QXZIadRbI5IET.dgndJgKQxdCYTg55C"
}

-------------------------------------------------------------------------------
{"level":"info","ts":1692986590.7736893,"logger":"logging_event_handler","msg":"[playbook debug]","name":"default","namespace":"service-telemetry","gvk":"infra.watch/v1beta1, Kind=ServiceTelemetry","event_type":"runner_on_ok","job":"6320470104503553430","EventData.TaskArgs":""}

--------------------------- Ansible Task StdOut -------------------------------

TASK [servicetelemetry : Create or patch htpasswd secret for grafana admin] ****
task path: /opt/ansible/roles/servicetelemetry/tasks/component_grafana.yml:72

-------------------------------------------------------------------------------
```

Some extra checking to see that everything is in alignment after setting `graphing.enabled: true`

```bash
oc get secret/default-grafana-htpasswd deployment/grafana-deployment -oyaml | grep -E '(hash|auth)'
    auth: cm9vdDokMnkkMTIkMjhuMnNpMnBSc2ZZMURyMFVRT0xHLlFYWklhZFJiSTVJRVQuZGduZEpnS1F4ZENZVGc1NUM=
      hash-of-creds-to-force-restart-if-changed: cm9vdDokMnkkMTIkMjhuMnNpMnBSc2ZZMURyMFVRT0xHLlFYWklhZFJiSTVJRVQuZGduZEpnS1F4ZENZVGc1NUM=
          hash-of-creds-to-force-restart-if-changed: cm9vdDokMnkkMTIkMjhuMnNpMnBSc2ZZMURyMFVRT0xHLlFYWklhZFJiSTVJRVQuZGduZEpnS1F4ZENZVGc1NUM=
          - -pass-basic-auth=false
          - -htpasswd-file=/etc/proxy/htpasswd/auth
          image: quay.io/openshift/origin-oauth-proxy:latest
          name: oauth-proxy
```

Then changing the password, I saw that Grafana restarted, and the output shows this:

```bash
    auth: cm9vdDokMnkkMTIkMjhuMnNpMnBSc2ZZMURyMFVRT0xHLmg5WDlGdHk4SWZxNEQ5cXVjNUVVZUF4STZDc25DSU8=
      hash-of-creds-to-force-restart-if-changed: cm9vdDokMnkkMTIkMjhuMnNpMnBSc2ZZMURyMFVRT0xHLlFYWklhZFJiSTVJRVQuZGduZEpnS1F4ZENZVGc1NUM=
          hash-of-creds-to-force-restart-if-changed: cm9vdDokMnkkMTIkMjhuMnNpMnBSc2ZZMURyMFVRT0xHLmg5WDlGdHk4SWZxNEQ5cXVjNUVVZUF4STZDc25DSU8=
          - -pass-basic-auth=false
          - -htpasswd-file=/etc/proxy/htpasswd/auth
          image: quay.io/openshift/origin-oauth-proxy:latest
          name: oauth-proxy
```

File: roles/servicetelemetry/tasks/component_grafana.yml [hunk 2]
```diff
@@ -7,8 +7,43 @@
...
   no_log: true
+  k8s_info:
+    api_version: v1
+    kind: Secret
+    namespace: '{{ ansible_operator_meta.namespace }}'
+    name: '{{ ansible_operator_meta.name }}-grafana-htpasswd'
+  register: grafana_htpasswd_secret
+
+- block:
+    - name: Set current Grafana htpasswd salt
```
Comment (csibbitt) [inline roles/servicetelemetry/tasks/component_grafana.yml:20]: s/Set/Read/ or Parse or something showing that it's a read op not a write up

File: roles/servicetelemetry/tasks/component_grafana.yml [hunk 3]
```diff
@@ -7,8 +7,43 @@
...
+      no_log: true
+      set_fact:
+        grafana_htpasswd_salt: "{{ ((grafana_htpasswd_secret.resources[0].data.auth | b64decode).split('$')[-1])[0:22] }}"
+  rescue:
+    - name: Generate initial Grafana htpasswd bcrypt string
+      no_log: true
+      set_fact:
+        init_grafana_htpasswd_bcrypt_string: "{{ (servicetelemetry_vars.graphing.grafana.admin_password | password_hash('bcrypt') | replace('$2b$','$2y$', 1)) }}"
+
+    - name: Rescue Grafana htpasswd salt
```
Comment (csibbitt) [inline roles/servicetelemetry/tasks/component_grafana.yml:30]: Read newly generated Grafana htpasswd salt

**Review by csibbitt — APPROVED** [review]
I heard naming things is hard so I offered some suggestions. Code looks good, though, thank you!

**Comment by leifmadsen** [comment]
> I heard naming things is hard so I offered some suggestions. Code looks good, though, thank you!

I did another walk through and made these a bit more descriptive. Let me know if you like this.

**Review by csibbitt — APPROVED** [review]

**Pull request merged**